### ISSUE TITLE
Error using the app after downloading config (Haxx setup failed: 0xffffffff)

### ISSUE THREAD

**Issue by bendoll** [yellows8/ctr-httpwn#34]
Hey there, I'm trying to use httpwn in preparation for a system transfer to avoid updating to 11.2, but I keep getting this strange error. I can't seem to find any information or solutions about it online.

I'm using the version from Plail's guide, but I think the only change should be the server it looks for. Other users who have used the same version from the guide have not had this error.

The app seems to work at first, but after it downloads the config, I get this
```
Preparing the haxx . . .
Searching for the httpc sharedmem in physmem . . .
Writing the haxx to physmem . . .
Triggering the haxx . . .
httpcCloseContext returned 0xffffffff.
Haxx setup failed: 0xffffffff.
An error occurred.
```
Any idea on how I can solve this? I'm on an NA n3ds XL with the latest versions of A9LH and Luma.

Here's a screenshot of the full screen.
![scr_5_top_left](https://cloud.githubusercontent.com/assets/10557798/20248923/c224f192-a9bc-11e6-914d-5fcbb1f3b65b.png)

Thanks in advance for the help, and all your work for the community.

**Comment by yellows8** [comment]
What system-version?

**Comment by bendoll** [comment]
Currently running 11.1

**Comment by yellows8** [comment]
What entrypoint?

**Comment by bendoll** [comment]
I'm using the homebrew launcher loader .cia installed to the home menu.

**Comment by yellows8** [comment]
No idea tbh.

**Comment by yellows8** [comment]
You've already attempted running it again after rebooting multiple times, right?

**Comment by bendoll** [comment]
Yeah I did multiple times. Tried different wifi networks and even my phone hotspot and got the same result. Currently booting up another n3ds with the same firmware and cfw setup to try out.

**Comment by bendoll** [comment]
Still the same result on a completely different n3ds. :\ this is really strange.

**Comment by yellows8** [comment]
When I've used it, this never happened with the release versions...

**Comment by Plailect** [comment]
I have no other reports of this issue.

**Comment by bendoll** [comment]
I can't imagine what I could have done wrong. Maybe I made some minor mistake when I was setting up cfw on the two systems? I'll keep trying different things, in the meantime if either of you have any more ideas or questions to throw my way, I'd appreciate them. Thanks for the responses so far.

**Comment by jaminkle** [comment]
I have the same error msg. I've tried a few diff types of servers now too.

**Comment by bendoll** [comment]
Well after a few more restarts and multiple tries copying new files over, something different happened. The app froze at `downloading config` and I let it be for a while. After a few minutes it let me use the cached info and it was successful. Tested eshop and it works fine, tested restarting and running ctr-httpwn again and the download fails again, but the cached info works fine.

 ![scr_30_top_left](https://cloud.githubusercontent.com/assets/10557798/20250653/0113f8f4-a9df-11e6-829d-f9281d7eb07c.png)
I copied whatever files were in the `root\3DS\ctr-httpwn` folder to my other n3ds and it worked this way as well. Will be trying a system transfer later tonight.

**Comment by rctgamer3** [comment]
I'm having the same issue. I compared the cached yls8 server_config with Plailects hosted copy and Plailects copy has a newline before `<?xml version="1.0" encoding="UTF-8"?>` (also closing tags on a new line and spaces instead of tabs). Doesn't seem to be the cause but Firefox marks it as an invalid xml.

Yellows8:

```
<?xml version="1.0" encoding="UTF-8"?>
<config>
```

Plailect:

```

<?xml version="1.0" encoding="UTF-8"?><config>
```

**Comment by Plailect** [comment]
Dammit. 

I had to change `<?xml version="1.0" encoding="UTF-8"?>` to `<?php echo '<?xml version="1.0" encoding="UTF-8"?>'; ?>` on the server because `short_open_tag` was set to `off` on the server I'm using.

I bet the lack of newline is the issue; I've reverted that line and enabled `short_open_tag`.

**Comment by thyraxx** [comment]
I also have this problem, and it persists, I can't seem to get it working even with different config links, it keeps returning 0xffffffff, even after restarting many times, and tried after removing MenuHax (if I must believe people this COULD be a problem), but still doesn't work :( , although I used Hans to get into eShop, but still, would be great to get this working.

EDIT: Im on 11.0 E sys.
EDIT2: Okay, it's fixed for me since the original servers are back on again, although I thought changing the URL should've also fixed it.

**Comment by PublicMenace** [comment]
I get the same Issue. I'm on 11.2 and I'm also using the Homebrew launcher loader .cia installed to the home menu.

**Comment by titan098** [comment]
If you are using ctr-httpwn 1.2 on an N3DS and you are getting this error, if you **disable** Clock+L2 in the Luma3DS config menu then ctr-httpwn 1.2 should then work.

**Comment by gauravg11** [comment]
Was experiencing the same issue on 11.0 with an N3DS. Disabling Clock+L2 did the trick.
